src/util.rs

```diff
@@ -53,3 +55,92 @@ where
     }
     i
 }
+
+pub fn varint_len(l: usize) -> usize {
+    if l < 0x80 {
+        1
+    } else if l < (1usize << 14) {
+        2
+    } else if l < (1usize << 21) {
+        3
+    } else if l < (1usize << 28) {
+        4
+    } else if l < (1usize << 35) {
+        5
+    } else if l < (1usize << 42) {
+        6
+    } else if l < (1usize << 49) {
+        7
+    } else if l < (1usize << 56) {
+        8
+    } else if l < (1usize << 63) {
+        9
+    } else {
+        10
+    }
+}
+
+pub unsafe fn encode_varint_uncheck(bytes: &mut [MaybeUninit<u8>], mut u: u64) -> usize {
+    let mut p = bytes.as_mut_ptr();
+    let mut c = 0;
+    while u >= 0x80 {
+        (*(&mut *p).as_mut_ptr()) = (u as u8 & 0x7f) | 0x80;
+        p = p.add(1);
+        u >>= 7;
+        c += 1;
+    }
+    (*(&mut *p).as_mut_ptr()) = u as u8;
+    c + 1
+}
+
+pub unsafe fn decode_varint_uncheck(bytes: &[u8]) -> (u64, usize) {
```

Comment:
You can write it in safe rust by using for loop.

https://github.com/BusyJay/pecan/blob/d4a17231253cd6b304a15d9d0f65edd5a0bcceab/utils/src/codec.rs#L54-L84 is an example.

Reply:
I copied `codec.rs` and modified input parameter as `BytesMut`, plus adapting error type to agatedb.